ports/zephyr/zephyr_display.c

```diff
@@ -0,0 +1,351 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/display.h>
+#include <stdlib.h>
+
+#include "modzephyr.h"
+#include "zephyr_device.h"
+#include "py/runtime.h"
+
+#if defined(CONFIG_DISPLAY)
+
+#if DT_HAS_COMPAT_STATUS_OKAY(zephyr_displays)
+#define ENUMERATE_DISPLAY_DEVS(node_id, prop, idx) DEVICE_DT_GET(DT_PROP_BY_IDX(node_id, prop, idx)),
+
+static const struct device *zephyr_display_devices[DT_ZEPHYR_DISPLAYS_COUNT] = {
+    DT_FOREACH_PROP_ELEM(DT_COMPAT_GET_ANY_STATUS_OKAY(zephyr_displays), displays, ENUMERATE_DISPLAY_DEVS)
+};
+#elif DT_HAS_CHOSEN(zephyr_display)
+static const struct device *zephyr_display_devices[] = {
+    DEVICE_DT_GET(DT_CHOSEN(zephyr_display)),
+};
+#else
+static const struct device *zephyr_display_devices[1];
+#endif
+
+const mp_obj_type_t zephyr_display_type;
+
+typedef struct _zephyr_display_obj_t {
+    mp_obj_base_t base;
+    const struct device *dev;
+} zephyr_display_obj_t;
+
+static void zephyr_display_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    zephyr_display_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_printf(print, "Display(\"%s\")", self->dev->name);
+}
+
+static mp_obj_t zephyr_display_make_new(const mp_obj_type_t *type,
+    size_t n_args, size_t n_kw, const mp_obj_t *args) {
+    mp_arg_check_num(n_args, n_kw, 1, 1, false);
+    const struct device *dev;
+
+    if (mp_obj_get_type(args[0]) == &mp_type_str) {
```

Comment:
`mp_obj_is_type()` is more conventional in these cases.

Reply:
done